src/main/java/org/opensearch/knn/index/engine/KNNIndexContext.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.engine;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.opensearch.Version;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.index.engine.qframe.QuantizationConfig;
+import org.opensearch.knn.index.mapper.PerDimensionProcessor;
+import org.opensearch.knn.index.mapper.PerDimensionValidator;
+import org.opensearch.knn.index.mapper.VectorValidator;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class provides the context to build an index for ANN search. All configuration is resolved before c
+ * construction and
+ */
+public final class KNNIndexContext {
```

Comment:
doesn't seem like the right place. Consider moving it to `index.model` or `index.params `